### PR TITLE
feat: add cypress

### LINKED ISSUE #385: [FEATURE] Adding cypress tests to conference website
### Why do we need this improvement?

To have a proper end to end and component test workflow.

### How will this change help?

Help with testing feature breakage.

### Screenshots

_No response_

### How could it be implemented/designed?

It is being implemented in cypress.

### 🚧 Breaking changes

No

### 👀 Have you checked for similar open issues?

- [X] I checked and didn't find a similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/asyncapi/.github/blob/master/CONTRIBUTING.md)

### Are you willing to work on this issue?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by ashmit-coder**
**Related issue(s)**

Resolves #385

### CODE REVIEW

File: .github/workflows/cypress-test.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: Cypress End-to-End test
+
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
```
Comment (AnimeshKumar923) [inline .github/workflows/cypress-test.yml:9]: I think we only need to run the tests on the incoming changes via a new pull request, so we don't need this part

File: .github/workflows/cypress-test.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+jobs:
+  cypress-run:
+    runs-on: ubuntu-22.04
```
Comment (AnimeshKumar923) [inline .github/workflows/cypress-test.yml:13]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/workflows/cypress-test.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```
Comment (AnimeshKumar923) [inline .github/workflows/cypress-test.yml:16]:
```suggestion
        uses: actions/checkout@v4
```
how about v4?

File: .github/workflows/cypress-test.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      
+      - name: Use Node.js
+        uses: actions/setup-node@v3
```
Comment (AnimeshKumar923) [inline .github/workflows/cypress-test.yml:19]:
```suggestion
        uses: actions/setup-node@v4
```
v4?

File: .github/workflows/cypress-test.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        with:
+            node-version: '20'
+
+      - name: Cypress run
+        uses: cypress-io/github-action@v6
+        with:
+          build: npm run build
+          start: npm start
```
Comment (AnimeshKumar923) [inline .github/workflows/cypress-test.yml:27]: Don't know much about the cypress testing part. Better to follow the official docs and figure it out.

**Review by AnimeshKumar923 — CHANGES_REQUESTED** [review]
I've reviewed it to the best of my knowledge. Let me know what you think @ashmit-coder

File: .github/workflows/cypress-test.yml (hunk 1 above)
Comment (ashmit-coder) [reply .github/workflows/cypress-test.yml:9]: Sometimes there are direct pushes by the admins, so I guess lets keep it?

File: .github/workflows/cypress-test.yml (hunk 5 above)
Comment (ashmit-coder) [reply .github/workflows/cypress-test.yml:27]: Yeah I did write it from the official [docs](https://docs.cypress.io/guides/continuous-integration/github-actions). Do have a look at them if you want.

**Comment by ashmit-coder** [comment]
cc: @AnimeshKumar923

**Comment by AceTheCreator** [comment]
@ashmit-coder please resolve the conflict in this pr

**Comment by ashmit-coder** [comment]
done @AceTheCreator

File: .github/workflows/cypress-test.yml (hunk 1 above)
Comment (AnimeshKumar923) [reply .github/workflows/cypress-test.yml:9]: Is that so? I don't know.

@AceTheCreator Can the bot push code directly to the main branch without first creating a branch?

File: .github/workflows/cypress-test.yml (hunk 5 above)
Comment (AnimeshKumar923) [reply .github/workflows/cypress-test.yml:27]: Okay :+1:

File: .github/workflows/cypress-test.yml (hunk 1 above)
Comment (AceTheCreator) [reply .github/workflows/cypress-test.yml:9]: Bot can't push code directly to main branch without creating PR first.

Comment (ashmit-coder) [reply .github/workflows/cypress-test.yml:9]: okay then this issue is resolved ig!

**Comment by ashmit-coder** [comment]
Hey @AceTheCreator , the review is completed. The PR is ready to merge!

**Review by Mayaleeeee — APPROVED** [review]

**Comment by thulieblack** [comment]
@ashmit-coder please resolve the conflicts

**Comment by ashmit-coder** [comment]
> @ashmit-coder please resolve the conflicts

Done @thulieblack

**Comment by thulieblack** [comment]
@AceTheCreator your review please

**Review by AceTheCreator — APPROVED** [review]
LGTM! well done @ashmit-coder

**Pull request merged**